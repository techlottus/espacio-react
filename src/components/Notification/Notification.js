import React, { useEffect, useState } from "react";
import { notiObs, notiObsClose } from "../../observables/notificationObs";
import "./Notification.scss";
import { Feedback } from "../Feedback/Feedback";
import { feedbackNoti } from "../../constants/Notification.constant";

const Notification = () => {
  const [items, setItems] = useState([]);

  const handleItem = (data) => {
    return (
      <div className="notiitem" key={data.id}>
        <Feedback
          isNoti={true}
          data={{
            ...feedbackNoti,
            left: {
              name: data.icon,
              status: data.type,
            },
          }}
          text={data.text}
          html={() => {return}}
          onRight={() => {
            notiObsClose.next(data.id);
          }}
        />
      </div>
    );
  };

  const handleItems = () => {
    return (
      <div className="notiitems">
        {items.map((item) => {
          return handleItem(item);
        })}
      </div>
    );
  };

  useEffect(() => {
    notiObsClose.subscribe((id) => {
      setItems((items) => {
        return items.filter((item) => item.id !== id);
      });
    });

    notiObs.subscribe((info) => {
      let id = new Date().getTime();
      let timer = setTimeout(() => {
        setItems((items) => {
          return items.filter((item) => item.id !== id);
        });
        clearTimeout(timer);
      }, 60000);
      setItems((items) => {
        if (items.length === 3) {
          items.pop();
        }
        let newItems = [
          {
            icon: info.type.icon,
            type: info.type.type,
            text: info.text,
            id,
          },
          ...items,
        ];
        return newItems;
      });
    });
  }, []);

  return (
    <>
      <div className="noti-container">{handleItems()}</div>
    </>
  );
};

export default Notification;
