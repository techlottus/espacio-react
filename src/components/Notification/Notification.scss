@import '../../App.scss';

.noti-container {
  background-color: greenyellow;
  //position: fixed;
  //height: 1000vh;
  //width: 100vw;
  z-index: 19;
  //top: 0;
}

.notiitems {
  position: fixed;
  bottom: 0;
  width: calc(100% - 32px);
  //height: calc(100vh - 56px);
  z-index: 19;
  display: flex;
  flex-direction: column-reverse;
  margin: 0 pixeltorem(16px);
  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    width: calc(40% - 32px);
    margin: 0 pixeltorem(36px);
  }

  .notiitem {
    width: 100%;
    height: 56px;
    margin-bottom: pixeltorem(16px);
  }
  
}