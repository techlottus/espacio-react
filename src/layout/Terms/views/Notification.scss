@import '../../../App.scss';

.notification-container {
  align-items: center;
  display: flex;
   width: 100%;
  justify-content: space-between;
  padding: 1rem 0;
  border: 0.0625rem solid map-get($palette-neutral, n10);
  background-color: map-get($palette-neutral, n00);
  border-radius: 0.5rem;

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    width: calc(100% - 2rem);
  } 

  > .side {
    padding: 0 1rem;
    text-align: center;
    &.-normal {
      color: map-get($palette-neutral, n60);
    }
    > span {
      font-size: 1.5rem;
    }
  }

  > .content {
    > .content-text {
      width: 100%;
      min-height: 100%;
      font-size: pixeltorem(16px);
      &.isbold{
        font-weight: 700;
        font-size: pixeltorem(18px);
        line-height: pixeltorem(24px);
      }
    }
  }
}
