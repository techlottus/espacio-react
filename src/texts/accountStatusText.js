export const accountStatusText = {
  statusTitle: "Estado de cuenta y consultas de facturas",
  resumeTitle: "Resumen de adeudo",
  total: "Total",
  errorMessage:
    "Tuvimos dificultades para cargar tu información en este momento",
  emptyMessage: {
    congrats: "Felicidades, estás al día con tus pagos",
    continue: "Continúa transformando tu futuro",
  },
  images: { 
    user: "images/user.png",
    empty: "yeah.png",
  },
  btnModal: {
    delete: "Eliminar",
    cancel: "Cancelar",
  },
  btnDownload: {
    download: "Descargar Estado de cuenta",
  },
  btnInvoice: {
    checkInvoice: "Consulta tus facturas",
  },
  tableAccountStatus: [
    { title: "Clave" },
    { title: "Ciclo" },
    { title: "Periodo" },
    { title: "Adeudo" },
  ],

  periodDetailTitle: "Detalles por periodo",
  periodDetailDescription:
    "Selecciona el programa y el periodo para ver los detalles de los movimientos; se clasifican de la siguiente manera:",
  periodDetailBullets: {
    debt: "Con adeudo",
    payed: "Pagados",
    toPay: "Por pagar",
  },
  periodDetailSelects: {
    program: "Programa",
    period: "Periodo",
  },
  accountDetailPeriod: {
    title: "Periodo",
    icon: "calendar_today",
    debt: "Saldo del periodo",
  },
  accountDetailMovements: {
    title: "Movimientos",
  },
  tableInvoiceQuery: [
    { title: "Fecha de emisión" },
    { title: "Periodo" },
    { title: "Concepto" },
    { title: "Monto" },
    { title: "RFC" },
    { title: "Preview" },
    { title: "PDF" },
    { title: "XML" },
  ],
  tableRowInvoiceQuery: [
    { title: "07 Abril 2021" },
    { title: "Periodo 5" },
    { title: "Colegiatura" },
    { title: "$2405.50" },
    { title: "LRKO485767HF2" },
  ],
  accountStatusInvoiceQuery: {
    title: "Consulta de facturas",
    description:
      "Filtra por rango de fechas, selecciona el perido que deseas consultar y presiona descargar.",
  },
  btnInvoiceQuery: {
    download: "Descargar todo",
    clear: "Limpiar",
  },
  linkInvoiceQuery: "Ver más"
};
