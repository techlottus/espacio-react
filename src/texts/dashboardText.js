export const dashboardText = {
  menu: {
    aula: {
      emoji: "mortar-board",
      text: "Aula",
    },
    book: {
      emoji: "books",
      text: "Biblioteca",
    },
    helpCenter: {
      emoji: "thought-balloon",
      text: "Centro de Ayuda",
    },
    procedures: {
      emoji: "file-folder",
      text: "Trámites",
    },
    accountStatus: {
      emoji: "chart-with-upwards-trend",
      text: "Estado de Cuenta",
    },
    payment:  {
      emoji: "credit-card",
      text: "Pagos",
    },
    invoice: {
      emoji: "page-facing-up",
      text: "Facturas"
    },
    benefits: {
      emoji: "star",
      text: "Portal de Beneficios"
    },
    sede: {
      emoji: "notebook",
      text: "SeDe"
    },
    psychopedagogical: {
      emoji: "blue-heart",
      text: "Centro de Atención Psicopedagógica"
    },
    evaluateServices: {
      emoji: "ec ec-white-check-mark",
      text: "Evalúa los servicios"
    },
    registrationSubjects: {
      emoji: "ec ec-closed-book",
      text: "Inscribe tus materias"
    },
    logout: {
      emoji: "back",
      text: "Cerrar sesión",
    },
  },
  titleWelcome: "Hola",
  images: {
    cardHeader: "images/header.png",
    empty: "empty.png",
  },
  logo: {
    urlLogo: "logo.svg"
  },
  logos: "logo.svg",
  sectionDashboard: {
    titleNews: "Noticias de tu universidad",
    titleCards: "¿Qué quieres hacer hoy?",
  },
  playIcon: "play_circle_outline",
  cards: {
    one: {
      title: "Aula",
      icon: "school",
    },
    two: {
      title: "Pagos",
      icon: "credit_card",
    },
    three: {
      title: "Biblioteca",
      icon: "import_contacts",
    },
    four: {
      title: "Trámites",
      icon: "post_add",
    },
    five: {
      title: "Inscribe tus materias",
      icon: "add_circle_outline",
    },
  },
  footerText: {
    text: "©2022 Derechos reservados Lottus Education",
    privacy: "Aviso de privacidad y términos y condiciones",
  },
};