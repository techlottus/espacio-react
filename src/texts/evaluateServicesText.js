export const evaluateServicesText = {
    evaluateServicesTitle: "Evalua tus servicios digitales",
    modalSurvey: {
        title: "Evalua tu experiencia para realizar tu pago a través de Espacio UTC",
        text: "¿Que tan probable es que recomiendes los pagos en Espacio UTC?"
    }
}