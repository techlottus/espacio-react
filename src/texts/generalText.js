export const generalText = {
  modals: {
    modalExit: {
      icon: "help",
      modalTitlePayment: "¿Seguro que deseas salir de la sección de pagos?",
      modalTitleProcedure: "¿Seguro que deseas abandonar el trámite?",
      exitBtn: "Aceptar",
      cancelBtn: "Cancelar",
    },
  },
};
