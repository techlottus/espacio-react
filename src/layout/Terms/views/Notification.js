import React from 'react'

import './Notification.scss'
const Notification = ({icon, text, isBold}) => {
  return (
    <>
      <div className='notification-container'>
        <div className="side -normal">
          <span className="material-icons icons">{icon}</span>
        </div>
        <div className="content">
          <p className={`content-text ${isBold ? 'isbold' : null }`}>{text}</p>
        </div>
        <div className="side"></div>
      </div>
    </>
  )
}

export default Notification;