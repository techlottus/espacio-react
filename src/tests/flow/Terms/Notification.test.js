import React from 'react';
import { mount } from 'enzyme';
import Notification from '../../../layout/Terms/views/Notification';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import configureStore  from "redux-mock-store";
import thunk from "redux-thunk";
import { termsText } from '../../../texts/termsText';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  texts: {
    terms: {
      ...termsText
    }
  }
};

let store = mockStore(initState);

window.dataLayer = {
  push: jest.fn()
}

const icon = 'person';
const text = 'Hola1w';

const wrapper = mount(
  <Provider store={store}>
    <Notification
      icon={icon}
      text={text}
    />
  </Provider>
)

describe('Test <Notification/>',() => {

  beforeEach(() => {
    store = mockStore(initState);
  })

  test('init component',() => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should valid text ',() => {
    expect(wrapper.find('.icons').text()).toBe(icon);
    expect(wrapper.find('.content-text').text()).toBe(text);
  })
})