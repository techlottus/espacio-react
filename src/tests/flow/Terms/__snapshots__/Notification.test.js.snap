// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Notification/> init component 1`] = `
<div
  className="notification-container"
>
  <div
    className="side -normal"
  >
    <span
      className="material-icons icons"
    >
      person
    </span>
  </div>
  <div
    className="content"
  >
    <p
      className="content-text null"
    >
      Hola1w
    </p>
  </div>
  <div
    className="side"
  />
</div>
`;
